### PR TITLE
Changed DateInput tests to remove react-test-renderer

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DateInput tests to remove react-test-renderer. This is being done to prepare for a fix to a DateInput issue, updating the tests first.

#### Where should the reviewer start?

test

#### What testing has been done on this PR?

unit test, review snapshots

#### How should this be manually tested?

review snapshot changes

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

no

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DateInput/__tests__/DateInput-test.js [hunk 1]
```diff
@@ -369,7 +351,8 @@ describe('DateInput', () => {
 
   test(`buttonProps should pass props to Button 
   when not inline and no format`, () => {
-    const component = renderer.create(
+    window.scrollTo = jest.fn();
```
Comment (jcfilben) [inline src/js/components/DateInput/__tests__/DateInput-test.js:354]: Why was `window.scrollTo` included here? It is not affecting the test or snapshot

Comment (ericsoderberghp) [reply src/js/components/DateInput/__tests__/DateInput-test.js:354]: Drop calls it via `setFocusWithoutScroll`. In the older `renderer` approach, we weren't going as far rendering, so this didn't show up. You can see how we do the same thing in Drop-test, Select-test, etc.

Comment (jcfilben) [reply src/js/components/DateInput/__tests__/DateInput-test.js:354]: Ah okay thanks for the explanation

**Review by jcfilben — APPROVED** [review]

**Review by jlaluces123 — COMMENTED** [review]
Just as a general question, since there are some components that already moved away from using `react-test-renderer` like this one, are there plans to fully migrate away from it down the line?

**Comment by ericsoderberghp** [comment]
> Just as a general question, since there are some components that already moved away from using `react-test-renderer` like this one, are there plans to fully migrate away from it down the line?

Yes. We're not in a huge rush, but that's the direction. And, it's better to update the tests like this without having any corresponding changes to the component in the same PR. So, I have a DateInput fix coming and wanted to get the tests updated first.

**Pull request merged**